### DISCUSSION TITLE
Nested Stack fails due to not authorized to perform iam:PassRole

### DISCUSSION THREAD

**Discussion by shoaibahmed10p** [aws/aws-cdk#31829] [Q&A]
Hello,

I have a NestedStack embedded within a Stack. The stack successfully last deployed 3 months ago. I have recently started working on it again and when I try to deploy, I get the following error

`2:11:52 PM | CREATE_FAILED        | AWS::CloudFormation::Stack           | *NestedStackResource*
API: cloudFormation User: arn:aws:sts::****:assumed-role/****/AWSCloudFormation is not authorized to perform: iam:PassRole on resource: arn:aws:iam::****:role/**** with an explicit deny in an identity-based policy`

Now the policy has not been edited for more than a year. I do have an explicit deny for iam:PassRole. **If I remove the Nested Stack, the stack deploys successfully**. No changes to the stack as of the last successful deployment.

Has something changed recently ? Anybody else faced this ?

aws-cdk-lib: 2.118.0
aws-cdk: 2.118.0

**Comment by khushail** [comment]
@shoaibahmed10p , looks like your account does not have the required permissions. you could check in console, what permissions your account has and does that include the permissions mentioned in error.

I found a link to kind of similar issue, you could refer to [this one](https://stackoverflow.com/questions/51911382/cloudformation-is-not-authorized-to-perform-iampassrole-on-resource) and see if that resolves the issue for you.
